.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  release:
+    types: [published]
+    branches: [ main ]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-22.04, windows-2022 ]
+        include:
+          - os: ubuntu-22.04
+            platform: linux
+          - os: windows-2022
+            platform: windows
+
+    env:
+      VCPKG_ROOT: 'vcpkg'
+      VCPKG_BINARY_SOURCES: 'clear;nuget,GitHub,readwrite'
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up MSVC (Windows)
+      if: matrix.os == 'windows-2022'
+      uses: ilammy/msvc-dev-cmd@v1
+      with:
+          arch: x64
+
+    - name: Install dependencies (Linux)
+      if: matrix.os == 'ubuntu-22.04'
+      run: |
+        sudo apt update
+        sudo apt upgrade -y
+        sudo apt install --no-install-recommends build-essential libtbb-dev ninja-build doxygen graphviz
+
+    - name: Install VCPKG
+      run: |
+        git clone https://github.com/Microsoft/vcpkg.git
+        cd vcpkg
+        ./bootstrap-vcpkg.sh -disableMetrics
+
+    - name: Build HealthGPS (debug)
+      run: |
+        cmake --preset=${{ matrix.platform }}-debug -DBUILD_BENCHMARKS=ON
+        cmake --build --preset=debug-build-${{ matrix.platform }} --target=install 
+
+    - name: Build HealthGPS (release)
+      run: |
```

Comment:
Why are we building both the debug and release versions every single time?

Reply:
That's what they were doing before, so I kept it in.

In *theory* you could have build failures that only happen when you compile in debug mode rather than release or vice versa, but it's not particularly likely and it doubles the amount of time the CI spends building things. In my last job, we only build-tested in release mode and that worked fine.

I'd be inclined to open an issue for this and change it in another PR, so that this one is a purely non-functional change, but if we're happy that this is the correct course of action, we could just do it now.

@jamesturner246?

Reply:
The build process is currently not so long anyway, but happy to leave it for now.

Reply:
Cool. I've opened an issue for future reference: #122